### PR TITLE
[Cody GA Docs] Add Cody pricing and usage

### PR DESCRIPTION
**Pull request by MaedahBatool**
First draft for Cody pricing docs.

## Test plan

Docs added

## Preview 🤩
[Preview Link](https://docs.sourcegraph.com/@cody-pricing)

### CODE REVIEW

File: doc/cody/usage-and-pricing.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+Support for Cody Pro is available through Discord, ensuring prompt assistance and guidance. However, it's essential to note that the plan is limited to a specified *number of developers*. Finally, you'll have default support with Anthropic and StarCoder as the officially supported LLMs.
+
+### Upgrading from Pro to Enterprise
+
+To upgrade from Cody Pro to Cody Enterprise, you should [Contact Sales](https://sourcegraph.com/contact/request-info) and connect with one of our support assistants. They will help you set up your account and get you started with Cody Enterprise.
+
+## Enterprise
+
+Cody Enterprise is designed for enterprises prioritizing security and administrative controls, starting at a consumption-based pricing capped at $19 per user per month. You get additional capabilities like BYOLLM, supporting Single-Tenant and Self Hosted setups for flexible coding environments.
```
[the remark this answers is not in the document]
Comment (chenkc805) [reply doc/cody/usage-and-pricing.md:37]: CC: @ryphil to review this section

**Review by chenkc805 — CHANGES_REQUESTED** [review]
Left a bunch of comments, thanks for driving this! It's getting there 💪

File: doc/cody/usage-and-pricing.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+Cody provides three subscription plans: **Free**, **Pro**, and **Enterprise**. Each plan is aimed to cater to a diverse range of users, from individual projects to large-scale enterprises. Cody Free includes basic features, while the Pro and Enterprise plans offer additional advanced features and resources to meet varying user requirements.
+
+## Free
+
+The free plan is designed for individuals to get started with Cody. It comes with a generous set of features to enhance your coding experience. It includes **500 autocompletions per month**, covering both whole and multi-line suggestions. You also get **20 chats/commands** per month, allowing for effective communication and control.
+
+The free plan provides access to local context with keyword search and embeddings on open-source code via Sourcegraph.com. Additionally, you can benefit from private embeddings *up to a specified amount*. The free plan ensures local context utilization and allows you to seamlessly integrate Cody into your preferred client, whether it's VSCode, JetBrains, or Neovim. Finally, you'll have default support with Anthropic and StarCoder as the officially supported LLMs.
```
[the remark this answers is not in the document]
Comment (MaedahBatool) [reply doc/cody/usage-and-pricing.md:11]: What does "user lifetime" mean here? Can you please elaborate?

**Comment by MaedahBatool** [comment]
Hey @chenkc805, I would like to get another pass from you. Addressed all your feedback in the PR.

File: doc/cody/usage-and-pricing.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (chenkc805) [reply doc/cody/usage-and-pricing.md]: "User lifetime" means they have 200 mb that they can ever use throughout the life of the user. 

So if they embed 150 MB of code, then they can only embed another 50 MB of code. If they delete that 150 MB of code embedding, then they have 200 MB of code embeddings available again. 

If they embed repo 1 = 150 MB and repo 2 = 25 MB, then they have only 25 MB left for other embeddings.

**Review by chenkc805 — APPROVED** [review]

File: doc/cody/usage-and-pricing.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+Cody provides three subscription plans: **Free**, **Pro**, and **Enterprise**. Each plan is aimed to cater to a diverse range of users, from individual projects to large-scale enterprises. Cody Free includes basic features, while the Pro and Enterprise plans offer additional advanced features and resources to meet varying user requirements.
+
+## Free
+
+The free plan is designed for individuals to get started with Cody. It comes with a set of features to enhance your coding experience. It includes **500 autocompletion suggestions** per month, covering both whole and multi-line suggestions. You also get **20 chats/commands** per month with access to creating Custom Commands.
+
+The free plan provides access to local context with keyword search and embeddings on open-source code via Sourcegraph.com. You get 200 MB of embeddings with the ability to manage these from the user settings. The free plan ensures local context utilization and allows you to seamlessly integrate Cody into your preferred client, whether it's VSCode, JetBrains, or Neovim. Finally, you'll have default support with Anthropic and StarCoder as the officially supported LLMs.
```
Comment (taylorsperry) [inline doc/cody/usage-and-pricing.md:11]: @chenkc805 Is there an expectation that generating embeddings for local projects is gated behind Pro? As far as I know, all PLG users will be able to generate "app-less" embeddings for GA.

File: doc/cody/usage-and-pricing.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+Until February 2024, you can upgrade to Cody Pro for free. If you want to continue using Cody Pro after February 2024, you get that at a **monthly price of $9 per user per month**.
+
+>NOTE: We'll be sharing payment details for the Pro plan in the coming weeks.
+
+## Pro
+
+Cody Pro, designed for individuals or small teams at **$9 per user per month**, offers an enhanced coding experience beyond the free plan. It provides unlimited autocompletion suggestions, allowing users to streamline their coding process without restrictions. Chat and commands executions are also unlimited, allowing users to create custom workflows.
+
+The plan includes the ability to create an unlimited number of Custom Commands for personalized workflows. In addition to using the local context of your code to improve responses, you can embed up to **1 GB** of your private code for even better Cody responses that reflect a deep understanding of your own code.
```
Comment (taylorsperry) [inline doc/cody/usage-and-pricing.md:29]: Same comment as above. If this work is underway, I don't know about it.

File: doc/cody/usage-and-pricing.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+Support for Cody Pro is available through Discord, ensuring prompt assistance and guidance. However, it's essential to note that the Cody Pro plan is limited to **50 developers** per month. Finally, you'll have default support with Anthropic and StarCoder as the officially supported LLMs. Moreover, Pro users will get an LLM selection for chat only. These LLMs are Claude Instant 1.2, Claude 2, ChatGPT 3.5 Turbo, and ChatGPT 4 Turbo Preview.
```
Comment (taylorsperry) [inline doc/cody/usage-and-pricing.md:31]: Note that LLM selection is only available for VSC today.

File: doc/cody/usage-and-pricing.md (hunk 3 above)
Comment (taylorsperry) [reply doc/cody/usage-and-pricing.md:11]: Also, this is only supported in VSC right now.

Comment (chenkc805) [reply doc/cody/usage-and-pricing.md:11]: > Is there an expectation that generating embeddings for local projects is gated behind Pro?

No, users will be able to have embeddings in all tiers: Free, Pro, and Enterprise. They just get different amounts: Free = 200mb/user/lifetime, Pro = 1 GB/user/lifetime, Enterprise = some higher amount. 

Good callout on VS Code though, we should mention this in the docs. Ultimately line 11 should read something like:

The free plan provides access to local context with keyword search and embeddings on open-source code via Sourcegraph.com. You **also** get 200 MB of embeddings with the ability to manage these from the user settings **in VS Code, with JetBrains IDE support coming soon**. The free plan ensures local context utilization and allows you to seamlessly integrate Cody into your preferred client, whether it's VSCode, JetBrains, or Neovim. Finally, you'll have default support with Anthropic and StarCoder as the officially supported LLMs.

File: doc/cody/usage-and-pricing.md (hunk 4 above)
Comment (chenkc805) [reply doc/cody/usage-and-pricing.md:29]: >In addition to using the local context of your code to improve responses, you can embed up to **1 GB** of your private code for even better Cody responses that reflect a deep understanding of your own code **in VS Code, with JetBrains IDE support coming soon.**

**Comment by MaedahBatool** [comment]
Since I have the approval on this PR, I am merging this with the main GA docs release. Find it here: https://github.com/sourcegraph/sourcegraph/pull/58903

**Pull request closed without merging**